sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessor.cs

```diff
@@ -30,6 +29,21 @@ public class EventProcessor<T> where T : BasePartitionProcessor, new()
         /// <summary>The primitive for synchronizing access during start and close operations.</summary>
         private readonly SemaphoreSlim _runningTaskSemaphore = new SemaphoreSlim(1, 1);
 
+        /// <summary>The primitive for synchronizing access during start and set handler operations.</summary>
+        private readonly object _setHandlerMutex = new object();
```

Comment:
nit:  This isn't technically a mutex.   (lock uses a Monitor under the covers) and it's really a guard against state transition to/from Active, right?    That is just a consideration of the handlers?

Reply:
Correct. What about `_setHandlerGuard` or `_setHandlerLock`?

Reply:
I'll stick to `_setHandlerGuard` unless you have something against it.